components/inputs/input-search.js

```diff
@@ -0,0 +1,238 @@
+			'ko': { 'search': '검색', 'search.clear': '검색 지우기' },
+			'nl': { 'search': 'Zoeken', 'search.clear': 'Zoekopdracht wissen' },
+			'pt': { 'search': 'Pesquisar', 'search.clear': 'Limpar Pesquisa' },
+			'sv': { 'search': 'Sökning', 'search.clear': 'Rensa sökning' },
+			'tr': { 'search': 'Ara', 'search.clear': 'Aramayı Temizle' },
+			'zh': { 'search': '搜索', 'search.clear': '清除搜索' },
+			'zh-tw': { 'search': '搜尋', 'search.clear': '清除搜尋' }
+		};
+	}
+
+	constructor() {
+		super();
+		this._focussed = false;
+		this._hovered = false;
+		this._lastSearchValue = '';
+		this.disabled = false;
+		this.noClear = false;
+		this.value = '';
+	}
+
+	get lastSearchValue() { return this._lastSearchValue; }
+	set lastSearchValue(val) {}
+	_setLastSearchValue(val) {
+		const oldVal = this._lastSearchValue;
+		this._lastSearchValue = val;
+		this.requestUpdate('lastSearchValue', oldVal);
+	}
+
+	render() {
+		requestAnimationFrame(
+			() => this.dispatchEvent(new CustomEvent('d2l-input-search-render', {
+				bubbles: false,
+				composed: false
+			}))
+		);
+		const inputClasses = {
+			'd2l-input': true,
+			'd2l-input-focus': !this.disabled && (this._focussed || this._hovered)
+		};
+		const showSearch = this._computeShowSearch();
+		return html`
+			<div class="d2l-input-search-container"
+				@mouseout="${this._handleMouseLeave}"
+				@mouseover="${this._handleMouseEnter}">
+				<input
+					aria-label="${ifDefined(this.label)}"
+					class="${classMap(inputClasses)}"
+					?disabled="${this.disabled}"
+					@input="${this._handleInput}"
+					@keypress="${this._handleInputKeyPress}"
+					maxlength="${ifDefined(this.maxlength)}"
+					placeholder="${ifDefined(this.placeholder)}"
+					type="search"
+					.value="${this.value}">
+				<d2l-button-icon
+					class="d2l-input-search-search"
+					?disabled="${this.disabled}"
+					?hidden="${!showSearch}"
+					icon="tier1:search"
+					@click="${this.search}"
+					text="${this.localize('search')}"></d2l-button-icon>
+				<d2l-button-icon
+					class="d2l-input-search-clear"
+					@click="${this._handleClearClick}"
+					?disabled="${this.disabled}"
+					?hidden="${showSearch}"
+					icon="tier1:close-default"
+					text="${this.localize('search.clear')}"></d2l-button-icon>
+			</div>
+		`;
+	}
+
+	connectedCallback() {
+		super.connectedCallback();
+		if (this.value !== undefined && this.value !== null) {
+			this._setLastSearchValue(this.value);
+		}
+		this.addEventListener('blur', this._handleBlur);
+		this.addEventListener('focus', this._handleFocus);
+	}
+
+	disconnectedCallback() {
+		this.removeEventListener('blur', this._handleBlur);
+		this.removeEventListener('focus', this._handleFocus);
+	}
+
+	focus() {
+		const elem = this.shadowRoot.querySelector('.d2l-input');
+		if (elem) elem.focus();
+	}
+
+	search() {
+		this._setLastSearchValue(this.value);
+		this._dispatchEvent();
+		if (!this.noClear && this.value.length > 0) {
+			this.updateComplete.then(() => {
```

Comment:
Nit: the `then` could be avoided here.
```
async search() {
	this._setLastSearchValue(this.value);
	this._dispatchEvent();
	if (!this.noClear && this.value.length > 0) {
		await this.updateComplete;
		this.shadowRoot.querySelector('.d2l-input-search-clear').focus();
	}
}
```

Reply:
`search()` is a public API on the component, so I didn't really want to change it... would that be a problem?

Reply:
Yeah, technically it results in a change to the return result, so probably good to leave it "as-is".